packages/vite/src/node/optimizer/index.ts

```diff
@@ -32,6 +32,7 @@ export type ExportsData = ReturnType<typeof parse> & {
   // es-module-lexer has a facade detection but isn't always accurate for our
   // use case when the module has default export
   hasReExports?: true
+  jsxLoader: boolean
```

Comment:
```suggestion
  // hint if the dep requires loading as jsx
  jsxLoader?: true
```

Perhaps the type should be like `hasReExports` to be consistent. I also added some docs for it.